src/main/resources/lib/unsplash/index.ts

```diff
@@ -1,76 +1,57 @@
-import { getSiteConfig } from "/lib/xp/portal";
-import { request } from "/lib/http-client";
-
-interface SearchPhotosParams {
-  query: string;
-  start: number;
-  count: number;
-}
-
-export interface UnsplashSearchResults {
-  id: string;
-  description: string;
-  alt_description: string;
-  user: {
-    name: string;
-  };
-  urls: {
-    thumb: string;
-  };
+import {
+  UnsplashImageMetadata,
+  UnsplashImageUrlParams,
+} from "/lib/unsplash/types";
+import { lookUpImageMetadata, storeImageMetadata } from "/lib/unsplash/repo";
+import { getPhotoById } from "/lib/unsplash/client";
+import { serviceUrl } from "/lib/xp/portal";
+import { notEmpty } from "/lib/unsplash/utils";
+export { searchPhotos } from "/lib/unsplash/client";
+
+export function getUnsplashImagesMetadata(
+  unsplashIds: string[]
+): UnsplashImageMetadata[] {
+  return unsplashIds.map(getUnsplashImageMetadata).filter(notEmpty);
 }
 
-interface UnsplashResponseBody {
-  total: number;
-  total_pages: number;
-  results: UnsplashSearchResults[];
-}
-
-export function searchPhotos({
-  query,
-  count,
-  start,
-}: SearchPhotosParams): UnsplashResponseBody {
-  const unsplashPageNumber = Math.floor(start / count);
-
-  const accessKey = getSiteConfig().unsplashAccessKey;
-  const url = "https://api.unsplash.com/search/photos";
-  const response = request({
-    url,
-    headers: {
-      Authorization: `Client-ID ${accessKey}`,
-    },
-    params: {
-      query,
-      page: unsplashPageNumber.toString(),
-      per_page: count.toString(),
-    },
+export function getUnsplashImageUrl(
+  unsplashId: string,
+  width: number,
+  height: number
+) {
+  const rawImageUrl = getUnsplashImageMetadata(unsplashId)?.urls.raw ?? "";
```

Comment:
I'm not a fan of using `""` as a fallback value. I think we might fool ourselves sometimes by masking an error case. I think `null` or `undefined` is clearer values to use when metadata doesn't exist.

```suggestion
  const rawImageUrl = getUnsplashImageMetadata(unsplashId)?.urls.raw;
```